src/gui/state.rs

```diff
@@ -202,6 +204,9 @@ impl UiState {
 
     /// Helper to strip out any ID-specific patterns from a label.
     #[inline]
+    // 🦀 Clippy lints that &self here goes unused, however when I remove it, it gives an error,
+    // could it be because of #[inline]? And if yes, how much will it hurt if we remove #[inline]?
+    #[allow(clippy::unused_self)]
```

Comment:
Yeah the reason this takes &self right now is because functionality is incomplete for it. The intent is that down the road, labels will be nested under parent containers - so additional bookkeeping will be required to know which parent a UI element is nested under. At present though, there is no nesting. Another reason why blindly denying and obeying lints is not always best practice.

It errors because all the usages of this call `self.get_label()` and so removing it would mean all those references would need to change to `get_label()`